autogen/agentchat/contrib/text_analyzer_agent.py

```diff
@@ -0,0 +1,72 @@
+from autogen import oai
+from autogen.agentchat.agent import Agent
+from autogen.agentchat.assistant_agent import ConversableAgent
+from typing import Callable, Dict, Optional, Union, List, Tuple, Any
+
+system_message = """You are an expert in text analysis.
+The user will give you TEXT to analyze.
+The user will give you analysis INSTRUCTIONS copied twice, at both the beginning and the end.
```

Comment:
Why INSTRUCTIONS copied twice?

Reply:
Great question! Empirically, it worked better on my tests than giving the instructions only once, whether before or after the text. And conceptually, this is what I expected would happen. The instructions often (but not always) consist of two parts: the description of the analysis, followed by the required format of the response. If the text appears first in the message, then the vector columns that process the text know nothing about the analysis to be performed. But if the analysis tokens come first, then the text vector columns are conditioned directly on the analysis tokens and vector columns, which seems like a more efficient use of transformer computation. But there are also benefits to having the analysis tokens appear for a second time at the end. They give the LLM more tokens to reason over, and force it to revisit the instructions. This is most useful as the text to be analyzed grows in length. Finally, it's important to have any output format requirements at the very end, so they don't get ignored.

Reply:
Nice trick!